### PR TITLE
#3689 error when no schema

### LINKED ISSUE #3689: Error if no JSONSchema
## Describe the bug

The validator throws an error if the schema is undefined

Screen Shot 2021-03-08 at 10 29 30 AM

### To reproduce

1. Don't have any schemas in db
2. Navigate to one of the schema dependent pages

### Expected behaviour

Don't bother validating if the schema is `undefined`

### Proposed Solution

Leave if you don't know how to fix/implement. Edit this issue description and explain here if you know the best path of implementing the fix within the codebase.

### Version and device info

- App version:
- Tablet model:
- OS version:

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #3689 

## Change summary

Fixes the error which is thrown if there's no schema.
Also removed a couple of warnings
- picker deprecation
- canSaveForm

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] To test the schema, I removed from desktop and cleared the app data. Set up mobile store again and tried to dispense vaccine

### Related areas to think about

If there are any general areas of the codebase your changes might have side affects on, mention them here

### CODE REVIEW

File: src/actions/Entities/NameNoteActions.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ export const NAME_NOTE_ACTIONS = {
 };
 
 const validateData = (jsonSchema, data) => {
+  if (!jsonSchema) return true;
```
Comment (mark-prins) [inline src/actions/Entities/NameNoteActions.js:23]: the validator throws an error if the schema is null

File: src/widgets/DropDown.js [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Picker } from 'react-native';
+import { Picker } from '@react-native-picker/picker';
```
Comment (mark-prins) [inline src/widgets/DropDown.js:9]: removing the deprecation warning

File: src/widgets/FormInputs/FormDateInput.js [hunk 3]
```diff
@@ -76,12 +76,9 @@ export const FormDateInput = React.forwardRef(
       const newValidity = onValidate(newValue);
       const updatedIsValid = validity && newValidity;
 
-      let updatedDate = null;
-      if (!newValue) {
-        updatedDate = null;
-      } else if (updatedIsValid) {
-        updatedDate = newDate.toDate();
-      }
+      const updatedDate = newValue && updatedIsValid ? newDate.toDate() : null;
```
Comment (mark-prins) [inline src/widgets/FormInputs/FormDateInput.js:79]: the date validation wasn't working, in that you could enter an invalid value, the field warns you, but the form `canSave` does not get set, as a `null` value is handed back to the form, which the form interprets as valid

File: src/widgets/JSONForm/JSONForm.js [hunk 4]
```diff
@@ -130,6 +130,8 @@ export const JSONForm = React.forwardRef(
     { formData, onChange, theme = defaultTheme, children, options, onSubmit, surveySchema },
     ref
   ) => {
+    if (!surveySchema) return null;
```
Comment (mark-prins) [inline src/widgets/JSONForm/JSONForm.js:133]: complaints about `forwardRef` being passed an invalid value for `value`

File: src/widgets/Tabs/PatientEdit.js [hunk 5]
```diff
@@ -78,8 +78,6 @@ const PatientEditComponent = ({
             shouldAutoFocus={false}
           />
         </View>
```
Comment (mark-prins) [inline src/widgets/Tabs/PatientEdit.js:80]: there are two vertical separators 🙄

File: src/actions/Entities/NameNoteActions.js (hunk 1 above)
Comment (joshxg) [reply src/actions/Entities/NameNoteActions.js:23]: 🤔 How does this even occur? Thought we had `{ surveySchema && <JSONForm/> }` everywhere

File: src/widgets/DropDown.js (hunk 2 above)
Comment (joshxg) [reply src/widgets/DropDown.js:9]: 🙇

File: src/widgets/FormInputs/FormDateInput.js (hunk 3 above)
Comment (joshxg) [reply src/widgets/FormInputs/FormDateInput.js:79]: 👍 

OK if you say so 🤣 , this component confuses me 😡

File: src/widgets/JSONForm/JSONForm.js (hunk 4 above)
Comment (joshxg) [reply src/widgets/JSONForm/JSONForm.js:133]: ahhhh

File: src/widgets/JSONForm/widgets/Select.js [hunk 6]
```diff
@@ -1,7 +1,9 @@
 /* eslint-disable react/forbid-prop-types */
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Picker } from 'react-native';
+// eslint-disable-next-line import/no-unresolved
+import { Picker } from '@react-native-picker/picker';
```
Comment (joshxg) [inline src/widgets/JSONForm/widgets/Select.js:5]: Not getting this linter warning for this one. Checkbox I was getting cause they seem to have screwed up their package json https://github.com/react-native-checkbox/react-native-checkbox/issues/111

**Review by joshxg — APPROVED** [review]
Amazing 👍

File: src/widgets/JSONForm/widgets/Select.js (hunk 6 above)
Comment (mark-prins) [reply src/widgets/JSONForm/widgets/Select.js:5]: hm. it was complaining.. it's fine now though. thanks

**Pull request merged**